src/storage/store/table_copy_utils.cpp

```diff
@@ -507,6 +508,13 @@ std::unique_ptr<arrow::PrimitiveArray> TableCopyUtils::createArrowPrimitiveArray
     return std::make_unique<arrow::PrimitiveArray>(type, length, buffer);
 }
 
+void TableCopyUtils::validateStrLen(uint64_t strLen) {
+    if (strLen > BufferPoolConstants::PAGE_4KB_SIZE) {
+        throw CopyException(
+            ExceptionMessage::overLargeStringValueException(std::to_string(strLen)));
+    }
+}
+
```

Comment:
```suggestion
void TableCopyUtils::validateStrLen(uint64_t strLen) {
    if (strLen > BufferPoolConstants::PAGE_4KB_SIZE) {
        throw CopyException(stringFormat("Maximum length of strings is 4096. Input string's length is {}.", strLen))
    }
}
```

Reply:
I see, but that exception message is also used by update function, when doing string updates

Reply:
I'd suggest refactoring `update()` to use `validateStrLen`.

Reply:
No, storage side shouldn't use this function. This function throws copyException while the storage should throw a runTimeException

Reply:
Then just change the argument to an integer instead of a string and call it.

Reply:
I don't understand what's the different between passing an int and a string?
If we pass an int, we also need to convert it to string inside the overLargeStringValueException.